spatialdata/_core/_spatialdata.py

```diff
@@ -27,57 +26,57 @@
 class SpatialData(IoMixin):
     """Spatial data structure."""
 
-    tables: Optional[AnnData]
-    labels: Optional[Mapping[str, Any]]
-    images: Optional[Mapping[str, Any]]
-    points: Optional[Mapping[str, AnnData]]
-    shapes: Optional[Mapping[str, AnnData]]
-    elems: List[Optional[Set[str]]]
+    tables: Optional[Mapping[str, AnnData]] = None
+    labels: Optional[Mapping[str, Any]] = None
+    images: Optional[Mapping[str, Any]] = None
+    points: Optional[Mapping[str, AnnData]] = None
+    elems: Set[str]
 
     def __init__(
         self,
-        tables: Optional[AnnData] = None,
+        tables: Optional[Mapping[str, AnnData]] = None,
         labels: Optional[Mapping[str, Any]] = MappingProxyType({}),
         labels_transform: Optional[Mapping[str, Any]] = None,
         images: Optional[Mapping[str, Any]] = MappingProxyType({}),
         images_transform: Optional[Mapping[str, Any]] = None,
-        points: Optional[Mapping[str, AnnData]] = None,
+        points: Optional[Mapping[str, AnnData]] = MappingProxyType({}),
         points_transform: Optional[Mapping[str, Any]] = None,
-        shapes: Optional[Mapping[str, AnnData]] = None,
-        shapes_transform: Optional[Mapping[str, Any]] = None,
     ) -> None:
 
         elems = []
 
         images, images_transform, elem_images = _validate_dataset(images, images_transform)
         labels, labels_transform, elem_labels = _validate_dataset(labels, labels_transform)
         points, points_transform, elem_points = _validate_dataset(points, points_transform)
-        shapes, shapes_transform, elem_shapes = _validate_dataset(shapes, shapes_transform)
 
-        if images is not None and images_transform is not None:
+        if TYPE_CHECKING:
+            assert images_transform is not None
+            assert labels_transform is not None
+            assert points_transform is not None
+
+        if images is not None:
             self.images = {
                 k: self.parse_image(image, image_transform)
                 for k, (image, image_transform) in zip(images.keys(), zip(images.values(), images_transform.values()))
             }
             elems.append(elem_images)
 
-        if labels is not None and labels_transform is not None:
+        if labels is not None:
             self.labels = {
                 k: self.parse_image(labels, labels_transform)
                 for k, (labels, labels_transform) in zip(labels.keys(), zip(labels.values(), labels_transform.values()))
             }
             elems.append(elem_labels)
 
-        if points is not None and points_transform is not None:
+        if points is not None:
             self.points = {k: self.parse_tables(points[k], points_transform[k]) for k in points.keys()}
```

Comment:
I checked all the changes, all good. I will just rename parse_tables to parse_points